Client/Frontend/Browser/Playlist/Managers & Cache/PlaylistMimeTypeDetector.swift

```diff
@@ -0,0 +1,314 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import MobileCoreServices
+
+// IANA List of Audio types: https://www.iana.org/assignments/media-types/media-types.xhtml#audio
+// IANA List of Video types: https://www.iana.org/assignments/media-types/media-types.xhtml#video
+// APPLE List of UTI types: https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html
+
+public class PlaylistMimeTypeDetector {
```

Comment:
Also can we test the detecting / caching functionality working properly in playlist after the changes ?

Reply:
I agree with the testing, but I didn't really change anything internally it's really a copy paste as is, no modification whatsoever. I just wanted to avoid scrolling down below this file every time I need to look at playlistwebloader.